mad-dog/src/emitter.js

```diff
@@ -62,20 +62,27 @@ class EmitterBasedTest {
   }
 
   /**
-   * Begin the test.
+   * Run test.
    * Returns when all inflight requests and ticks are completed.
    */
-  async start () {
+  async run () {
     logger.info('Beginning ticking.')
     this.isTicking = true
     await tick(this.emitter, this.tickIntervalSeconds, this.testDurationSeconds)
     this.isTicking = false
     logger.info('Finished ticking.')
 
+    const withTimeout = async (requestPromise, timeoutMs, timeoutMsg) => {
```

Comment:
nit - can we construct the err msg in the fn `withTImeout` instead of passing it in? also, can we rename this to something more descriptive, like `executeFnWithTimeout`?  

```
    const withTimeout = async (requestPromise, timeoutMs) => {
      const timeoutPromise = new Promise((resolve, reject) => {
        setTimeout(() => reject(new Error(`Failed to resolve all inflight requests in ${timeoutMs}ms`)), timeoutMs)
      })
      return Promise.race([requestPromise, timeoutPromise])
    }

    // Await any pending requests if necessary
    if (this.inFlightCount > 0) {
      logger.info(`Awaiting [${this.inFlightCount}] inflight requests.`)
      await withTimeout(this.inFlightPromise, 300000)
    }
    logger.info('Test done.')
  }
```